site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md

```diff
@@ -0,0 +1,396 @@
+and group names.
+
+## Identity transformations and policies
+
+When a user authenticates, the configuration of the OIDCIdentityProvider, ActiveDirectoryIdentityProvider, or 
+LDAPIdentityProvider resource determines how the user's username and group names will be extracted from the external
+identity provider in a protocol-specific way (e.g. via OIDC ID token claims or LDAP record attributes).
+
+Then, operating on the username and group names extracted from the external IDP:
+- Identity **transformations** can change either the user's username or group names.
+- Identity **policies** can reject the user's authentication based on their username and/or groups.
+
+Identity transformations and policies are configured on the FederationDomain, so they are specific to that specific
+FederationDomain's usage of the external identity provider.
+
+Transformations and policies are configured using the Common Expression Language (CEL) programming language.
+They are configured as a list, and they will be executed in the order specified. The output of each transformation
+or policy expression may impact the input values for the next expression from the list.
+
+Pinniped's implementation of CEL expressions includes the
+[standard language features](https://github.com/google/cel-spec/blob/master/doc/langdef.md)
+as well as [the string extensions](https://github.com/google/cel-go/tree/master/ext#strings).
+
+### Pipelines of identity transformation and policy `expressions`
+
+There are three types of transformation expressions:
+- `username/v1` are expressions which may change the user's username. These expressions must return a string,
+  and the value of the string will be the user's username. Returning an empty string or a string that contains
+  only whitespace characters will cause an authentication error. Returning the value of the `username` variable
+  unmodified will leave the username unchanged.
+- `groups/v1` are expressions which may change the group names of the groups to which the user belongs.
+  These expressions must return a list of strings. The returned list may be empty. The returned list will
+  be the names of the groups to which the user belongs. Returning the value of the `groups` variable
+  unmodified will leave the username unchanged.
+- `policy/v1` are expressions which may reject the user's authentication based on their username and/or groups.
+  These expressions must return a boolean. Returning true have no impact on the user's authentication
+  and will therefore allow the user's authentication to continue. Returning true also has no impact
+  on the username or group names. Returning false will the reject user's authentication
+  and the user will see the error message configured for that policy expression (or a default error message).
+  Rejecting a user's authentication prevents the user from authenticating into every cluster
+  which chooses to use this FederationDomain for identity services. This happens before
```

Comment:
perhaps "which uses this FederationDomain"?

Reply:
Updated